docs/security.md

```diff
@@ -149,30 +149,44 @@ secret file agrees with the executors' secret file.
 
 # Network Encryption
 
-Spark supports two mutually exclusive forms of encryption for RPC connections.
+Spark supports two mutually exclusive forms of encryption for RPC connections:
 
-The first is an AES-based encryption which relies on a shared secret, and thus requires
-RPC authentication to also be enabled.
+The **preferred method** uses TLS (aka SSL) encryption via Netty's support for SSL. Enabling SSL
+requires keys and certificates to be properly configured. SSL is standardized and considered more
+secure.
 
-The second is an SSL based encryption mechanism utilizing Netty's support for SSL. This requires
-keys and certificates to be properly configured. It can be used with or without the authentication
-mechanism discussed earlier.
-
-One may prefer to use the SSL based encryption in scenarios where compliance mandates the usage
-of specific protocols; or to leverage the security of a more standard encryption library. However,
-the AES based encryption is simpler to configure and may be preferred if the only requirement
-is that data be encrypted in transit.
+The legacy method is an AES-based encryption mechanism relying on a shared secret. This requires
+RPC authentication to also be enabled. This method uses a bespoke protocol and it is recommended
+to use SSL instead.
```

Comment:
Looking at the earlier text, we are dropping some verbiage from it - any particular reason for it ?
Specifically `"However, the AES based encryption is simpler to configure and may be preferred if the only requirement is that data be encrypted in transit."` for ex.

We have made our recommendation clear, and rest should be left to users to determine - and make the right tradeoff.

Reply:
I will take this out. I think it is rationalizing why there is an ad hoc protocol here instead in the first place. I agree that we should make it clear that TLS is strongly preferred and make it easy to migrate.

Reply:
I put the original text back in.